src/api.ts

```diff
@@ -14,18 +22,72 @@ router.get('/', async (req, res) => {
 router.get('/test', async (req, res) => {
   const url: any = req.query.url || '';
   const event = {
-    id: `proposal/0x38c654c0f81b63ea1839ec3b221fad6ecba474aa0c4e8b4e8bc957f70100e753`,
-    space: 'pistachiodao.eth',
+    id: `proposal/0xc59f05d899cd80178300b724c4bf43a037a908741369690243aa4a77bbafbf18`,
+    space: 'fabien.eth',
     event: 'proposal/created',
-    expire: 1647343155
+    expire: 1664272182
   };
   try {
-    new URL(url);
-    await sendEvent(event, url);
+    // new URL(url);
+    // await sendEvent(event, url);
+    await sendPushNotification(event);
     return res.json({ url, success: true });
   } catch (e) {
     return res.json({ url, error: e });
   }
 });
 
+router.post('/device', async (req, res) => {
+  const owner = req.body?.owner;
+  const token = req.body?.token;
+  try {
+    // Add token to DB and subscribe user to topics
+    await subscribeUser(token, owner);
+    return res.json({ success: true });
+  } catch (error) {
+    console.log('[notifications] Error adding device', error);
+    return sendError(res, 'Error enabling push notifications');
+  }
+});
+
+router.delete('/device', async (req, res) => {
+  const owner = req.body?.owner;
+  const token = req.body?.token;
+  try {
+    // Delete token from DB and unsubscribe user from topics
+    await unsubscribeUser(token, owner);
+    return res.json({ success: true });
+  } catch (error) {
+    console.log('[notifications] Error deleting device', error);
+    return sendError(res, 'Error disabling push notifications');
+  }
+});
+
+router.post('/subscribed', async (req, res) => {
+  const owner = req.body?.owner;
+  const token = req.body?.token;
+  try {
+    // Validate token from DB and subscribe user to topics
+    const subscribed = await isTokenValid(token, owner);
+    return res.json({ subscribed });
+  } catch (error) {
+    console.log('[notifications] Error validating device', error);
+    return sendError(res, 'Error validating device');
+  }
+});
+
+router.post('/subscription', async (req, res) => {
```

Comment:
This route assume the user join / leave a space using snapshot.org but he may activate notification on snapshot.org but actually join or leave from another integration that wont trigger this route. Also this has no authentication anyone could exploit this route and subscribe any user to a specific space. We should instead use the code that replay the messages from the hub to check when there is a new follow or unfollow action. Similar than what we do to populate proposal events.